include/extractor/traffic_signals.hpp

```diff
@@ -22,6 +23,14 @@ struct TrafficSignals
         return bidirectional_nodes.count(to) > 0 || unidirectional_segments.count({from, to}) > 0;
     }
 };
+
+struct TrafficSignals final : public RoadObjects {};
+
+// TODO: better naming ?
```

Comment:
It feels like we can try to generalise the whole this logic to any kind of objects which follows the same "idea" as traffic lights, i.e. some nodes which can be directed and give additional penalty in routing.

Reply:
Maybe `ControlFlow`?

If we view them as node penalties applicable by traversal direction, then I can see them being part of the same logical grouping.